src/Ellipsoid.jl

```diff
@@ -159,3 +171,46 @@ function ∈(x::AbstractVector{N}, E::Ellipsoid{N})::Bool where {N<:AbstractFloa
     w, Q = x-E.center, E.shape_matrix
     return dot(w, Q \ w) ≤ 1
 end
+
+"""
+    rand(::Type{Ellipsoid}; [N]::Type{<:Real}=Float64, [dim]::Int=2,
+         [rng]::AbstractRNG=GLOBAL_RNG, [seed]::Union{Int, Nothing}=nothing
+        )::Ellipsoid{N}
+
+Create a random ellipsoid.
+
+### Input
+
+- `Ellipsoid` -- type for dispatch
+- `N`         -- (optional, default: `Float64`) numeric type
+- `dim`       -- (optional, default: 2) dimension
+- `rng`       -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`      -- (optional, default: `nothing`) seed for reseeding
+
+### Output
+
+A random ellipsoid.
+
+### Algorithm
+
+The center is a normally distributed vector with entries of mean 0 and standard
+deviation 1.
+
+The idea for the shape matrix comes from
+[here](https://math.stackexchange.com/a/358092).
+The matrix is symmetric positive definite, but also diagonally dominant.
```

Comment:
Describe with an equation?

```math
Q =  \frac{1}{2}(S + S^T) + nI,
```
where `n = dim` (defaults to 2), and ``S `` is a ``n\\times n`` randomly matrix with coefficients uniformly distributed in the interval `[-1, 1]`.